test/test_generation.py

```diff
@@ -123,3 +123,53 @@ def test_fourier_synthesis_linescan_hrms_more_wavevectors():
     #                         - np.mean(realised_rms_heights))**2)))
     assert abs(np.mean(realised_rms_heights) -
                ref_height) / ref_height < 0.1  #
+
+
+def test_fourier_synthesis_3D():
+    n = [101, 101, 101]
+    nx, ny, nz = n
+    s = [101, 101, 101]
+    sx, sy, sz = s
+    g = [sx/nx, sy/ny, sz/nz]
+    gx, gy, gz = g
+    hurst = 0.8
+    rms_height = 0.5
+
+    x = np.arange(0, sx, gx)
+    y = np.arange(0, sy, gy)
+    z = np.arange(0, sz, gz)
+    X, Y = np.meshgrid(x, y)
+
+    c = fourier_synthesis((nx, ny, nz), (sx, sy, sz),
+                          hurst, rms_height=rms_height,
+                          long_cutoff=sx / 3.)
+
+    hursts = np.zeros(nx)
+    rmss = np.zeros(nx)
+    for i in range(nx):
+        physical_sizes = np.array([sy, sz])
+        topo = c[i, :, :]
+        rmss[i] = rms(topo)
+
+    mean_rms_x = np.mean(rmss)
+    assert(((mean_rms_x - rms_height) / rms_height) < 0.5)
+
+    hursts = np.zeros(ny)
+    rmss = np.zeros(ny)
+    for i in range(ny):
+        physical_sizes = np.array([sx, sz])
+        topo = c[:, i, :]
+        rmss[i] = rms(topo)
+
+    mean_rms_y = np.mean(rmss)
+    assert(((mean_rms_y - rms_height) / rms_height) < 0.5)
+
+    hursts = np.zeros(nz)
+    rmss = np.zeros(nz)
+    for i in range(nz):
+        physical_sizes = np.array([sx, sy])
+        topo = c[:, :, i]
+        rmss[i] = rms(topo)
+
+    mean_rms_z = np.mean(rmss)
+    assert(((mean_rms_z - rms_height) / rms_height) < 1e-5)
```

Comment:
I don't understand why it should be more accurate on xy planes. Or did you forget to reduces the tolerance up there ?

Reply:
I did forget to change it